cpp/src/ffi/reader_writer_jni.cpp

```diff
@@ -0,0 +1,512 @@
+// Copyright 2023 Zilliz
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "milvus-storage/ffi_jni.h"
+#include "milvus-storage/ffi_c.h"
+#include <arrow/c/abi.h>
+#include <arrow/c/bridge.h>
+#include <cstring>
+#include <memory>
+#include <vector>
+
+// ==================== JNI Utility Functions ====================
+
+void ThrowJavaExceptionFromFFIResult(JNIEnv* env, const struct ffi_result* result) {
+  if (IsSuccess(const_cast<FFIResult*>(result))) {
+    return;
+  }
+
+  const char* message = GetErrorMessage(const_cast<FFIResult*>(result));
+  const char* exception_class = "java/lang/RuntimeException";
+
+  switch (result->err_code) {
+    case LOON_INVALID_ARGS:
+      exception_class = "java/lang/IllegalArgumentException";
+      break;
+    case LOON_MEMORY_ERROR:
+      exception_class = "java/lang/OutOfMemoryError";
+      break;
+    case LOON_ARROW_ERROR:
+    case LOON_LOGICAL_ERROR:
+    case LOON_GOT_EXCEPTION:
+    case LOON_UNREACHABLE_ERROR:
+    case LOON_INVALID_PROPERTIES:
+    default:
+      exception_class = "java/lang/RuntimeException";
+      break;
+  }
+
+  jclass exc_class = env->FindClass(exception_class);
+  if (exc_class != nullptr) {
```

Comment:
`assert exc_class != nullptr`? the exception should not be empty. 

if current exception is nullptr, it will cause new prolem